src/App.js

```diff
@@ -4,45 +4,596 @@
 import React from "react";
 // DOM bindings for React Router
 import { BrowserRouter as Router } from "react-router-dom";
+//> Additional
+// SHA Hashing algorithm
+import sha256 from "js-sha256";
+//> Intel
+import { Intel } from "snek-intel";
+//> MDB
+// "Material Design for Bootstrap" is a great UI design framework
+import { MDBProgress } from "mdbreact";
 
 //> Components
 /**
  * Footer: Global Footer
  * Navbar: Global navigation bar
  */
-//import { Footer, Navbar } from "./components/molecules";
-
+import { Footer, Navbar } from "./components/molecules";
+// Starts the page on top when reloaded or redirected
+import { ScrollToTop } from "./components/atoms";
 //> Routes
 import Routes from "./Routes";
 //#endregion
 
 //#region > Components
+/**
+ * @class Root component which loads all other components
+ */
 class App extends React.Component {
+  state = {
+    loggedUser: undefined,
+    fetchedUser: undefined,
+    loading: true,
+    caching: false,
+  };
+
+  componentDidMount = () => {
+    // Create new intel instance
+    this.intel = new Intel();
+    // Create new session link for easy access
+    this.session = this.intel.snekclient.session;
+    // Begin session
+    this.begin();
+  };
+
+  //> Authentication methods
+  /**
+   * Begin Session
+   *
+   * @description Start a new session based on authentication history.
+   *              New site access will lead to a anonymous login.
+   */
+  begin = async (user) => {
+    //#TSID1
+    //console.log("BEGIN INIT", user);
+    const whoami = await this.session.begin(user ? user : undefined);
+
+    // Check if whoami is not empty
+    if (whoami?.username || whoami?.whoami?.username) {
+      const username = whoami?.username
+        ? whoami?.username
+        : whoami?.whoami?.username;
+
+      // Check if whoami user is not anonymous user
+      if (username !== process.env.REACT_APP_ANONYMOUS_USER) {
+        // Get loggedUser object
+        const loggedUser = {
+          username,
+          avatarUrl:
+            "https://avatars2.githubusercontent.com/u/21159914?u=afab4659183999f1adc85089bb713aefbf085b94",
```

Comment:
I think the anonuser looks a lot like someone working at this project xD

```suggestion
            "https://www.clipartmax.com/png/full/166-1669056_the-20-cooler-octocat-github-octocat.png",
```

Reply:
or
```suggestion
            "https://anonymous-austria.com/src/img/logo.png",
```